basic_hashtable/b_hashtables.c

```diff
@@ -104,6 +127,15 @@ void hash_table_remove(BasicHashTable *ht, char *key)
  ****/
 char *hash_table_retrieve(BasicHashTable *ht, char *key)
 {
+  int index = hash(key, ht -> capacity);
+
+  if (ht -> storage[index] != NULL)
+  {
+    if (strcmp(ht -> storage[index] -> key, key) == 0)
```

Comment:
don't need this check - `if (ht -> storage[index] != NULL)` does all we need for this one